ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/Deduplicate.java

```diff
@@ -59,8 +62,22 @@ Jedis getJedis() {
    * Static factory methods.
    */
 
-  public static Deduplicate removeDuplicates(ValueProvider<URI> uri) {
-    return new RemoveDuplicates(uri);
+  /**
+   * Returns a {@link PTransform} that checks message IDs with Redis and discards any messages
+   * with IDs that have already been seen. The error collection contains messages with poorly-formed
+   * IDs or where Redis was not available.
+   */
+  public static //
+      PTransform<PCollection<PubsubMessage>, ResultWithErrors<PCollection<PubsubMessage>>> //
+      removeDuplicates(ValueProvider<URI> uri) {
+    return PTransform.compose((PCollection<PubsubMessage> input) -> {
+      ResultWithErrors<PCollection<PubsubMessage>> result = input.apply(new RemoveDuplicates(uri));
+      PCollection<PubsubMessage> errorsStrippedOfDuplicates = result.errors()
+          .apply(Filter.by((PubsubMessage message) -> !Optional
+              .ofNullable(message.getAttribute("exception_class"))
+              .filter(RemoveDuplicates.DuplicateIdException.class.getName()::equals).isPresent()));
+      return ResultWithErrors.of(result.output(), errorsStrippedOfDuplicates);
+    });
```

Comment:
I think it would be better to route duplicates to a separate output when caught, rather than filtering them out here.